liota/dccs/iotcc.py

```diff
@@ -214,9 +219,29 @@ def create_relationship(self, reg_entity_parent, reg_entity_child):
             if entity_obj.unit is not None:
                 self.publish_unit(reg_entity_child, entity_obj.name, entity_obj.unit)
         else:
+            def on_response(msg):
+                try:
+                    log.debug("Received msg: {0}".format(msg))
+                    json_msg = json.loads(msg)
+                    log.debug("Processing msg: {0}".format(json_msg["type"]))
+                    self._check_version(json_msg)
+                    if json_msg["type"] == "create_relationship_response" and json_msg["body"][
+                        "result"] == "succeeded" and json_msg["body"]["parent"] == reg_entity_parent.reg_entity_id and \
+                                    json_msg["body"]["child"] == reg_entity_child.reg_entity_id:
+                        log.info(
+                            "Relationship between entities {0} & {1} created successfully in IoTCC".format(
+                                reg_entity_parent.ref_entity.name, reg_entity_child.ref_entity.name))
+                    else:
+                        log.info("Waiting for create relationship response")
+                        on_response(self.recv_msg_queue.get(True, timeout))
+                except Exception as err:
+                    log.exception("Exception while create relationship request")
```

Comment:
Do we need to log as well as raise the err?

Reply:
The log messages provide more details on where exactly the exception occurs(in which request) rather than looking at the stack trace.